openquake/hazardlib/tests/gsim/mgmpe/modifiable_gmpe_test.py

```diff
@@ -228,6 +228,55 @@ def test_set_total_std_as_tau_plus_phi(self):
         # Original tau for PGA is 0.6201
         np.testing.assert_almost_equal(stddev[0], 0.5701491121, decimal=6)
 
+    def test_horiz_comp_to_geom_mean01(self):
+        """ Checks the horizontal component conversion """
+        gmpe_name = 'BooreEtAl2014'
+        gmm_unscaled = ModifiableGMPE(gmpe={gmpe_name: {}})
+        gmm = ModifiableGMPE(gmpe={gmpe_name: {}}, horiz_comp_to_geom_mean={})
+        stddevs = [const.StdDev.TOTAL]
+        expected = [1.009, 1.009, 1.0100744, 1.01714271, 1.0264865,
+                    1.03168663, 1.04649942, 1.077]
+        results = []
+        for imt in ([PGA(), SA(0.01), SA(0.1), SA(0.2), SA(0.5), SA(1), SA(5),
+                     SA(10)]):
+            res_unscaled = gmm_unscaled.get_mean_and_stddevs(
+                self.ctx, self.ctx, self.ctx, imt, stddevs)
+            mean_unscaled = res_unscaled[0]
+            res_scaled = gmm.get_mean_and_stddevs(
+                self.ctx, self.ctx, self.ctx, imt, stddevs)
+            mean = res_scaled[0]
+            results.append(np.exp(mean_unscaled[0]) / np.exp(mean[0]))
+        np.testing.assert_almost_equal(results, expected, decimal=5)
+
+    def test_horiz_comp_to_geom_mean02(self):
+        """ Checks the horizontal component conversion """
+        gmpe_name = 'BooreAtkinson2008'
+        gmm_unscaled = ModifiableGMPE(gmpe={gmpe_name: {}})
+        gmm = ModifiableGMPE(gmpe={gmpe_name: {}}, horiz_comp_to_geom_mean={})
+        stddevs = [const.StdDev.TOTAL]
+        expected = [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
+        imts = [PGA(), SA(0.01), SA(0.1), SA(0.2), SA(0.5), SA(1), SA(5),
+                SA(10)]
+        convs = [0.02, 0.03, 0.03, 0.0317185550924273, 0.0371922924372184,
+                 0.04, 0.04, 0.04]
+
+        results = []
+        for imt, cnv in zip(imts, convs):
+            res_unscaled = gmm_unscaled.get_mean_and_stddevs(
+                self.ctx, self.ctx, self.ctx, imt, stddevs)
+            mean_unscaled = res_unscaled[0]
+            std_unscaled = res_unscaled[1][0]
+            res_scaled = gmm.get_mean_and_stddevs(
```

Comment:
New code should not use  the old method `.get_mean_and_stddevs``and use the function `contexts.get_mean_stds` instead.